### PR TITLE
Win probability

### PR DESCRIPTION
**Pull request by NathanDeMaria**
Add a page for getting the win % for a matchup

### CODE REVIEW

File: web/src/containers/Matchup/MatchupDisplay/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    const { team1, team2, matchup } = matchupState;
+    return {
+        team1,
+        team2,
+        matchup,
+    }
+};
+
+export default connect(mapStateToProps)(MatchupDisplay as any); // TODO: typescript better
```
Comment (c-m-hunt) [inline web/src/containers/Matchup/MatchupDisplay/index.tsx]: I've been using this workaround which actually feels ok https://github.com/Microsoft/TypeScript/issues/25103#issuecomment-412806226

Comment (c-m-hunt) [reply web/src/containers/Matchup/MatchupDisplay/index.tsx]: Having said that, I'm not getting the error any more. Possibly fixed in TS?

File: web/tslint.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     ]
   },
   "rules": {
+    "no-console": false,
```
Comment (c-m-hunt) [inline web/tslint.json]: Remove after dev?

File: web/src/actions/actionCreators.ts [hunk 3]
```diff
@@ -116,6 +117,7 @@ export interface ITeamsPayload {
 
 export const getTeams = (league: string) => {
   return (dispatch: ThunkDispatch<IAppState, void, ITeamsPayload>) => {
+    // TODO: don't re-get if it's already in there
```
Comment (c-m-hunt) [inline web/src/actions/actionCreators.ts]: Quite simple to do here if you want:
```
  return (dispatch: ThunkDispatch<IAppState, void, ITeamsPayload>, getState: any) => {
    const { leagueTeams } = getState();
    // check league teams in object
```
But possibly better to put condition in components to only retrieve if teams props not populated.

....or maybe even better to get teams on first app load only and remove `getTeams` from all components?

File: web/src/containers/Matchup/MatchupDisplay/index.tsx (hunk 1 above)
Comment (NathanDeMaria) [reply web/src/containers/Matchup/MatchupDisplay/index.tsx]: Yeah looks like this just works now, 🤷‍♂ I had been copy pasting the boilerplate for new components 😳

File: web/src/containers/Matchup/TeamSelector/TeamSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  private getMatchup = () => {
+    const selected = this.teamsAreSelected();
+    if (selected) {
+      const { team1, team2 } = selected;
+      this.props.getMatchup(team1, team2);
+    }
+  }
+
+  private teamsAreSelected = (): ISelectedTeams | null => {
+    if (this.state.team1 !== null && this.state.team2 !== null) {
```
[the remark this answers is not in the document]
Comment (NathanDeMaria) [reply web/src/containers/Matchup/TeamSelector/TeamSelector.tsx]: I might want to allow that, hopefully this page eventually becomes more than one `<p>` with the probability

File: web/src/actions/actionCreators.ts (hunk 3 above)
Comment (NathanDeMaria) [reply web/src/actions/actionCreators.ts]: On first app load sounds like the right move, everything's going to use that. Same with "leagues". What's the good way to do that? Connect the [`App`](https://github.com/NathanDeMaria/py-glicko/blob/e2f69c955f47128ba6f0f2398fc662d3af673878/web/src/App.tsx#L49)? Add some code [here in the root `index`](https://github.com/NathanDeMaria/py-glicko/blob/ea2c735a9b88011af87a24a3b5063d8a424e3367/web/src/index.tsx#L11)?

Comment (c-m-hunt) [reply web/src/actions/actionCreators.ts]: Yeah that's where I've done it. If you've got the "loading" notifications in the components anyway, should only need putting in there and removing the load from all components

**Pull request merged**